FreeRTOS/Test/CMock/message_buffer/message_buffer_utest.c

```diff
@@ -0,0 +1,670 @@
+                                                int cmock_num_calls )
+{
+    uint8_t data[ TEST_MAX_MESSAGE_SIZE ] = { 0 };
+    size_t dataReceived = 0;
+
+    /* Read next message from message buffer to wake up sender task. */
+    dataReceived = xMessageBufferReceive( xMessageBuffer, data, TEST_MAX_MESSAGE_SIZE, 0 );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, dataReceived );
+    return pdTRUE;
+}
+
+static BaseType_t messageBufferSendFromISRCallback( UBaseType_t uxIndexToWaitOn,
+                                                    uint32_t ulBitsToClearOnEntry,
+                                                    uint32_t ulBitsToClearOnExit,
+                                                    uint32_t * pulNotificationValue,
+                                                    TickType_t xTicksToWait,
+                                                    int cmock_num_calls )
+{
+    BaseType_t receiverTaskWokenFromISR = pdFALSE;
+    uint8_t data[ TEST_MAX_MESSAGE_SIZE ] = { 0 };
+    size_t dataSent = 0;
+
+    dataSent = xMessageBufferSendFromISR( xMessageBuffer, data, TEST_MAX_MESSAGE_SIZE, &receiverTaskWokenFromISR );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, dataSent );
+    return pdTRUE;
+}
+
+static BaseType_t messageBufferSendCallback( UBaseType_t uxIndexToWaitOn,
+                                             uint32_t ulBitsToClearOnEntry,
+                                             uint32_t ulBitsToClearOnExit,
+                                             uint32_t * pulNotificationValue,
+                                             TickType_t xTicksToWait,
+                                             int cmock_num_calls )
+{
+    uint8_t data[ TEST_MAX_MESSAGE_SIZE ] = { 0 };
+    size_t dataSent = 0;
+
+    /* Send a single message of max message size to message buffer to wake up receiver Task. */
+    dataSent = xMessageBufferSend( xMessageBuffer, data, TEST_MAX_MESSAGE_SIZE, 0 );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, dataSent );
+    return pdTRUE;
+}
+
+static BaseType_t receiverTaskNotificationCallback( TaskHandle_t xTaskToNotify,
+                                                    UBaseType_t uxIndexToNotify,
+                                                    uint32_t ulValue,
+                                                    eNotifyAction eAction,
+                                                    uint32_t * pulPreviousNotificationValue,
+                                                    int cmock_num_calls )
+{
+    TEST_ASSERT_EQUAL( receiverTask, xTaskToNotify );
+    receiverTaskWoken++;
+    return pdTRUE;
+}
+
+/*******************************************************************************
+ * Unity fixtures
+ ******************************************************************************/
+void setUp( void )
+{
+    assertionFailed = 0;
+    xMessageBuffer = NULL;
+    senderTaskWoken = 0;
+    receiverTaskWoken = 0;
+    shouldAbortOnAssertion = pdTRUE;
```

Comment:
Please call the _Init function for each of your mocked files. Otherwise, the test cases are tightly coupled and failures could cascade.
```
mock_task_Init();
mock_fake_assert_Init();
mock_fake_port_Init();
```